File: etablissementAngular/src/app/model/utilisateur.ts
import { Adresse } from "./adresse";
import { Etablissement } from "./etablissement";
import { Login } from "./login";

export class Utilisateur {
    private _id!: number;
    private _login!: Login;
    private _nom: string;
    private _prenom: string;
    private _adresse!: Adresse;
    private _dateNaissance: string;
    private _etablissement!: Etablissement;


	constructor(id?: number, login?: Login, nom: string='', prenom: string='', adresse?: Adresse, dateNaissance: string='', etablissement?: Etablissement) {
		
		
		this._nom = nom;
		this._prenom = prenom;
		this._dateNaissance = dateNaissance;
		
	}


    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Getter login
     * @return {Login}
     */
	public get login(): Login {
		return this._login;
	}

    /**
     * Getter nom
     * @return {string}
     */
	public get nom(): string {
		return this._nom;
	}

    /**
     * Getter prenom
     * @return {string}
     */
	public get prenom(): string {
		return this._prenom;
	}

    /**
     * Getter adresse
     * @return {Adresse}
     */
	public get adresse(): Adresse {
		return this._adresse;
	}

    /**
     * Getter dateNaissance
     * @return {string}
     */
	public get dateNaissance(): string {
		return this._dateNaissance;
	}

    /**
     * Getter etablissement
     * @return {Etablissement}
     */
	public get etablissement(): Etablissement {
		return this._etablissement;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}

    /**
     * Setter login
     * @param {Login} value
     */
	public set login(value: Login) {
		this._login = value;
	}

    /**
     * Setter nom
     * @param {string} value
     */
	public set nom(value: string) {
		this._nom = value;
	}

    /**
     * Setter prenom
     * @param {string} value
     */
	public set prenom(value: string) {
		this._prenom = value;
	}

    /**
     * Setter adresse
     * @param {Adresse} value
     */
	public set adresse(value: Adresse) {
		this._adresse = value;
	}

    /**
     * Setter dateNaissance
     * @param {string} value
     */
	public set dateNaissance(value: string) {
		this._dateNaissance = value;
	}

    /**
     * Setter etablissement
     * @param {Etablissement} value
     */
	public set etablissement(value: Etablissement) {
		this._etablissement = value;
	}



}

File: etablissement/src/main/java/com/projetfinal/etablissement/entity/Utilisateur.java
package com.projetfinal.etablissement.entity;

import java.time.LocalDate;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Transient;
import javax.persistence.Version;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Entity
@SequenceGenerator(name = "seqUtilisateur", sequenceName = "seq_utilisateur", initialValue = 10, allocationSize = 1)
public class Utilisateur {
	
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqUtilisateur")
	private Integer id;
	@NotNull
	private Login login;
	@NotEmpty
	private String nom, prenom;
	@Transient
	@NotNull
	private Adresse adresse;
	@NotNull
	private LocalDate dateNaissance;
	@NotNull
	@ManyToOne
	private Etablissement etablissement;
	@Version
	private int version;
}
